package com.alix.tsuki.settings.sources.manage.plugins.model

import androidx.annotation.StringRes
import com.alix.tsuki.list.ui.model.ListModel

sealed interface PluginManageItem : ListModel {

	data class Plugin(
		val name: String,
		val repository: String?,
		val installedTag: String?,
		val latestTag: String?,
	) : PluginManageItem {

		val displayName: String
			get() = name.removeSuffix(".jar")

		val hasUpdate: Boolean
			get() = !latestTag.isNullOrBlank() && latestTag != installedTag

		override fun areItemsTheSame(other: ListModel): Boolean {
			return other is Plugin && name == other.name
		}
	}

	data class Placeholder(
        @field:StringRes val titleResId: Int,
        @field:StringRes val summaryResId: Int?,
	) : PluginManageItem {
		override fun areItemsTheSame(other: ListModel): Boolean {
			return other is Placeholder && titleResId == other.titleResId && summaryResId == other.summaryResId
		}
	}
}
